src/foam/util/FluentSpec.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * @license
+ * Copyright 2022 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  extends: 'foam.core.FObjectProperty',
+  
+  package: 'foam.util',
+  name: 'FObjectSpec',
+
+  documentation: `
+    A Property which stores knowledge of what FObject to create
+    rather than storing an instance of an FObject.
+  `,
+
+  properties: [
+    [
+      'fromJSON',
+      function fromJSON(value, ctx, prop, json) {
+        /** Prevents specs from instantiating when loaded from JSON. **/
+        return value;
+      }
+    ],
+    [ 'adapt', function(_, spec, prop) {
+      return foam.String.isInstance(spec) ? { class: spec } : spec ;
+    } ],
+    [ 'javaJSONParser', 'foam.lib.json.UnknownFObjectParser.instance()' ],
+    [ 'displayWidth', 80 ]
+  ]
+});
+
+foam.CLASS({
+  extends: 'foam.core.FObjectArray',
```

Comment:
for consistency placing the extends axiom below package and name is more uniform and just easier to read.

Reply:
Consistency is good, but I'm going to refer back to something I keep trying to convince people: being too strict about code style means that ideas which could be expressed by code style are lost.

This example proposes the interesting idea that `extends` of a class definition should be on top for the same reason that `class` is at the top of a property. One could think of the value for `extends` as the _type_ of a class. Is that a good way to think about it? Well, now I have you wondering that, and you wouldn't have been wondering that if it didn't land in this PR.

Reply:
Also if ever someone wanted to normalize everything and remove all these inconsistencies from all files it could be done using the existing javascript builder with some configuration tweaks.

Reply:
I decided in this case it's not that important and changed it anyway. Just because it's interesting though I wanna add two more things
- maybe having `extends` after is only easier to read because of convention (same order as Java for example)
- I noticed some chaining in foam is like `this().` instead of `.that()`. I enjoyed finding that inconsistency because it made me think about the code in a different way.